kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/LocallyRunningOperatorRbacHandler.java

```diff
@@ -0,0 +1,242 @@
+ * <br/>
+ * Register this extension before LocallyRunOperatorExtension, passing LocallyRunOperatorExtension
+ * the kubernetes client returned by {@link #operatorClient}.  This client will use a principal
+ * which has been configured in Kubernetes to have the same RBAC rights as the operator will use
+ * in production.
+ * <br/>
+ * For the test's interactions with Kubernetes, where more privileges are required, use the operations
+ * exposed by {@link #testActor(AbstractOperatorExtension)}.
+ *
+ */
+public class LocallyRunningOperatorRbacHandler implements BeforeEachCallback, AfterEachCallback {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LocallyRunningOperatorRbacHandler.class);
+
+    private final String impersonatedUser = UUID.randomUUID().toString();
+
+    @AutoClose
+    private final KubernetesClient operatorClient = OperatorTestUtils.kubeClient(
+            new KubernetesClientBuilder().editOrNewConfig().withImpersonateUsername(impersonatedUser).endConfig());
+
+    @AutoClose
+    private final KubernetesClient testActorClient = OperatorTestUtils.kubeClient();
+
+    // @formatter:off
+    private final ClusterRole frameworkClusterRole = new ClusterRoleBuilder()
+            .withNewMetadata()
+                .withName("frameowork-cluster-role")
+            .endMetadata()
+            .addNewRule()
+                .addToApiGroups("")
+                .addToVerbs("get", "create", "delete", "patch")
+                .addToResources("namespaces")
+            .endRule()
+            .addNewRule()
+                .addToApiGroups("apiextensions.k8s.io")
+                .addToVerbs("get", "list", "watch", "create", "delete", "patch", "delete")
+                .addToResources("customresourcedefinitions")
+                .endRule()
+            .build();
+    // @formatter:on
+
+    // @formatter:off
+    private final ClusterRoleBinding frameworkClusterRoleBinding = new ClusterRoleBindingBuilder()
+            .withNewMetadata()
+                .withName("frameowork-cluster-role-binding")
+            .endMetadata()
+            .withNewRoleRef()
+                .withName(frameworkClusterRole.getMetadata().getName())
+                .withKind(frameworkClusterRole.getKind())
+                .withApiGroup("rbac.authorization.k8s.io")
+            .endRoleRef()
+            .addNewSubject()
+                .withName(impersonatedUser)
+                .withKind("User")
+                .withApiGroup("rbac.authorization.k8s.io")
+            .endSubject()
+            .build();
+    // @formatter:on
+
+    @Override
+    public void beforeEach(ExtensionContext context) {
+        try (var adminClient = OperatorTestUtils.kubeClient();
+                var files = Files.list(Path.of("install"))) {
+            files.sorted().forEach(resourceFile -> {
+                try {
+                    var list = adminClient.load(new FileInputStream(resourceFile.toString())).items();
+                    list.stream()
```

Comment:
*  We only really need to know the (cluster) roles to be used when testing the given reconciler. This might be a subset of all the roles needed for the operator as a whole (i.e. I think its useful to track the permissions at the level of the reconciler/controller and run the test with only those).
* If we know the roles it's easy to construct a binding to the impersonated user. i.e. we don't really need to load or identify the binding(s) from the manifests.
* Hard-coding a directory of manifests is not ideal. The files to load should be configurable some how. Maybe using a glob to make it easier to identify the manifests (e.g. `*-ClusterRole-*.yaml` thus not selecting any bindings).

Reply:
pushed this up @tombentley, you now configure it with the resource dir and clusterRole globs. It loads ClusterRoles only and synthesizes a CRB per role.

Reply:
also the framework role and binding are created/destroyed using the same mechanism, so we don't have special fields for them any more.